core/tracker/tracker.go

```diff
@@ -134,11 +141,57 @@ func analyseDutyFailed(duty core.Duty, es []event) (bool, component, string) {
 // TODO(xenowits): Implement logic for reporting duties.
 func failedDutyReporter(core.Duty, bool, string, string) {}
 
-// analyseParticipation returns the share indexes of peers that participated in this duty.
-// TODO(dhruv): implement logic to analyse participation.
-//nolint:deadcode
-func analyseParticipation(core.Duty, []event) []int {
-	return nil
+// analyseParticipation returns a set of share indexes of participated peers.
+func analyseParticipation(events []event) map[int]bool {
+	// Set of shareIdx of participated peers.
+	resp := make(map[int]bool)
+
+	for _, e := range events {
+		// If we get a parSigDBInternal event, then the current node participated successfully.
+		// If we get a parSigEx event, then the corresponding peer with e.shareIdx participated successfully.
+		if e.shareIdx > 0 && (e.component == parSigEx || e.component == parSigDBInternal) {
+			resp[e.shareIdx] = true
+		}
+	}
+
+	return resp
+}
+
+// newParticipationReporter returns a new participation reporter function which logs and instruments peer participation.
+func newParticipationReporter(peers []p2p.Peer) func(context.Context, core.Duty, map[int]bool) {
+	// lastParticipation is the set of peers who participated in the last duty.
+	lastParticipation := make(map[int]bool)
```

Comment:
nit: `prevAbsent := make(...)`

Reply:
can just make this `[]string` and compare against `fmt.Sprint(absent) != fmt.Sprint(prevAbsent)`